src/plugins/condition/ConditionManager.js

```diff
@@ -265,14 +270,25 @@ export default class ConditionManager extends EventEmitter {
         this.emit(`broadcastTelemetry`, Object.assign({}, this.createNormalizedDatum(datum, id), {id: id}));
     }
 
+    getTestData(metadatum) {
+        const found = this.testData.conditionTestInputs.find((testInput) => (testInput.metadata === metadatum.key) || (testInput.metadata === metadatum.source));
+        return found ? found.value : false;
+    }
+
     createNormalizedDatum(telemetryDatum, id) {
         return Object.values(this.telemetryObjects[id].telemetryMetaData).reduce((normalizedDatum, metadatum) => {
+            const value = this.testData.applied && this.getTestData(metadatum);
             const formatter = this.openmct.telemetry.getValueFormatter(metadatum);
-            normalizedDatum[metadatum.key] = formatter.parse(telemetryDatum[metadatum.source]);
+            normalizedDatum[metadatum.key] = value ?  formatter.parse(value) : formatter.parse(telemetryDatum[metadatum.source]);
             return normalizedDatum;
         }, {});
     }
 
+    updateTestData(testData) {
+        this.testData = testData;
+        this.openmct.objects.mutate(this.conditionSetDomainObject, 'configuration.conditionTestData', this.testData);
```

Comment:
Ah! Ok so the test data is literally available only at the time of edit and then flushed. I can work with that.

Reply:
Done.

Reply:
I feel very strongly that test data should NOT be flushed - don't throw away work the user has done. Instead,
- Test data should ONLY be applied when editing a Condition Set.
- Test data should never be enabled automatically when beginning to edit a Condition Set, especially if that section is collapsed by default. This means when edit ends, disable test data.

Reply:
@charlesh88 Right. Persisting test data is fine *so long as it is not being used for the evaluation of condition sets outside of the condition set edit view*. From looking at the code it seemed as though test values were being used wherever conditions are evaluated, but I may have misinterpreted it.

Reply:
I think we still want to make it *really* clear to the user when test data is being used in the evaluation, even in the Condition Set editor. Especially if it's being persisted, because it will be easy to forget.

Reply:
@shefalijoshi Have changes that address this been pushed? If so, I can sign off on this. If they haven't yet, we should not merge.